Bug: LocalStorage service type writes metadata with wrong permission

### What happened?

The issue was identified by RSTUF functional tests as reported here:
- https://github.com/vmware/repository-service-tuf/issues/222

The Worker container writes the file to the volume with permission 0o600 by default.

https://github.com/vmware/repository-service-tuf-worker/blob/3dd3f36cd4e63f18217e7d1da91dd568e3232957/repository_service_tuf_worker/services/storage/local.py#L84-L96

The HTTP container (running Apache) exposes the files using the www user. For Apache security, the files are not accessible, including readable by others, and the HTTP Clients get 403 which is forbidden.

The issue is unreproducible in the development environment because we use python as HTTP Server, which uses the same user and exposes it directly.

We don't need to consider permission mode 0o600 for the metadata files. It is a security requirement for private keys. The `LocalStorage` writes metadata files that are public and readable to others.

### What steps did you take?

1. Deploy RSTUF using Apache Server (httpd image) or NGNIX
2. The exposed metadata volume will not be readable by HTTP clients

### What behavior did you expect?

* Move the restrict default to False
* Remove the restrict and write the files

### Relevant log output

```log
=========================== short test summary info ============================
[3055](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3056)
FAILED tests/functional/targets/test_add_targets.py::test_adding_a_target_using_rstuf_api[630-"716f6e863f744b9ac22c97ec2f677b76ea5f5908bc5bc61510bfc4751384ea7a"-{"key": "value"}-"file1.tar.gz"] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3056](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3057)
FAILED tests/functional/targets/test_add_targets.py::test_adding_a_target_using_rstuf_api[2024-"93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc853eaa0ac8fa3b46f6e863f74"-None-"a/file2.tar.gz"] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3057](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3058)
FAILED tests/functional/targets/test_add_targets.py::test_adding_a_target_using_rstuf_api[532-"d9f34f8cd5cb3b3eb79b3e4b5dae3a16df499a70d8a053eaa0ac8fa5f5908bc5"-{"key": "value"}-"a/b/file3.tar.gz"] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3058](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3059)
FAILED tests/functional/targets/test_performance.py::test_api_requester_multiple_request_and_targets[2-2-10] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3059](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3060)
FAILED tests/functional/targets/test_performance.py::test_api_requester_multiple_request_and_targets[2-100-60] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3060](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3061)
FAILED tests/functional/targets/test_performance.py::test_api_requester_multiple_request_and_targets[5-10-30] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3061](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3062)
FAILED tests/functional/targets/test_performance.py::test_api_requester_multiple_request_and_targets[100-2-120] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3062](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3063)
FAILED tests/functional/targets/test_performance.py::test_api_requester_multiple_request_and_targets[50-50-240] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3063](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3064)
FAILED tests/functional/targets/test_remove_targets.py::test_removing_a_target_using_rstuf_api[["file1.tar.gz"]] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3064](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3065)
FAILED tests/functional/targets/test_remove_targets.py::test_removing_a_target_using_rstuf_api[["file1.tar.gz", "a/file2.tar.gz"]] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3065](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3066)
FAILED tests/functional/targets/test_remove_targets.py::test_removing_a_target_using_rstuf_api[["file1.tar.gz", "a/file2.tar.gz", "c/d/file3.tar.gz"]] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3066](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3067)
FAILED tests/functional/targets/test_remove_targets.py::test_removing_targets_that_does_exist_and_ignoring_the_rest[["file1.tar.gz"]-["foo"]] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
[3067](https://github.com/vmware/repository-service-tuf/actions/runs/4004879734/jobs/6874569753#step:15:3068)
FAILED tests/functional/targets/test_remove_targets.py::test_removing_targets_that_does_exist_and_ignoring_the_rest[["file1.tar.gz", "a/file2.tar.gz"]-["foo", "bar"]] - tuf.api.exceptions.DownloadHTTPError: 403 Client Error: Forbidden for url: http://localhost:8080/timestamp.json
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/repository_service_tuf_worker/interfaces.py
+++ b/repository_service_tuf_worker/interfaces.py
@@ -16,7 +16,7 @@
 
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
-from typing import Any, Dict, List, Optional
+from typing import Any, Dict, List
 
 from tuf.api.metadata import Metadata, T
 
@@ -89,7 +89,6 @@ class IStorage(ABC):
         self,
         file_data: bytes,
         filename: str,
-        restrict: Optional[bool] = True,
     ) -> None:
         """
         Stores file bytes within a file with a specific filename.

--- a/repository_service_tuf_worker/services/storage/local.py
+++ b/repository_service_tuf_worker/services/storage/local.py
@@ -4,10 +4,9 @@
 
 import glob
 import os
-import stat
 from contextlib import contextmanager
 from io import BufferedReader
-from typing import List, Optional
+from typing import List
 
 from securesystemslib.exceptions import StorageError  # noqa
 
@@ -73,7 +72,6 @@ class LocalStorage(IStorage):
         self,
         file_data: bytes,
         filename: str,
-        restrict: Optional[bool] = True,
     ) -> None:
         """
         Writes passed file object to configured TUF repo path using the passed
@@ -81,22 +79,8 @@ class LocalStorage(IStorage):
         """
         filename = os.path.join(self._path, filename)
 
-        if restrict:
-            # On UNIX-based systems restricted files are created with read and
-            # write permissions for the user only (octal value 0o600).
-            fd = os.open(
-                filename, os.O_WRONLY | os.O_CREAT, stat.S_IRUSR | stat.S_IWUSR
-            )
-        else:
-            # Non-restricted files use the default 'mode' argument of os.open()
-            # granting read, write, and execute for all users (mode 0o777).
-            # NOTE: mode may be modified by the user's file mode creation mask
-            # (umask) or on Windows limited to the smaller set of OS supported
-            # permisssions.
-            fd = os.open(filename, os.O_WRONLY | os.O_CREAT)
-
         try:
-            with os.fdopen(fd, "wb") as destination_file:
+            with open(filename, "wb") as destination_file:
                 destination_file.write(file_data)
                 destination_file.flush()
                 os.fsync(destination_file.fileno())

--- a/tests/unit/tuf_repository_service_worker/services/storage/test_local.py
+++ b/tests/unit/tuf_repository_service_worker/services/storage/test_local.py
@@ -2,8 +2,6 @@
 #
 # SPDX-License-Identifier: MIT
 
-import os
-
 import pretend
 import pytest
 
@@ -172,18 +170,8 @@ class TestLocalStorageService:
         ]
         assert local.glob.glob.calls == [pretend.call("/path/2.root.json")]
 
-    def _put_setup(self, fake_destination_file, restrict=True):
-        """Setup helper for all required functions in LocalStorage.put()"""
-
-        class FakeDestinationFile:
-            def __init__(self):
-                pass
-
-            def __enter__(self):
-                return fake_destination_file
-
-            def __exit__(self, type, value, traceback):
-                pass
+    def test_put(self, monkeypatch):
+        service = local.LocalStorage("/path")
 
         local.os = pretend.stub(
             path=pretend.stub(
@@ -192,65 +180,39 @@ class TestLocalStorageService:
                 )
             ),
             fsync=pretend.call_recorder(lambda *a: None),
-            O_WRONLY=0,
-            O_CREAT=0,
-            open=pretend.call_recorder(lambda *a: 0),
-            fdopen=pretend.call_recorder(lambda *a: FakeDestinationFile()),
         )
 
-        if restrict:
-            local.stat = pretend.stub(S_IRUSR=0, S_IWUSR=0)
-
-    def test_put(self):
-        service = local.LocalStorage("/path")
+        fake_file_data = b"fake_byte_data"
 
         fake_destination_file = pretend.stub(
-            write=pretend.call_recorder(lambda *a: None),
+            write=pretend.call_recorder(lambda *a: "bytes_data"),
             flush=pretend.call_recorder(lambda: None),
             fileno=pretend.call_recorder(lambda: "fileno"),
         )
 
-        fake_bytes = b"data"
+        class FakeDestinationFile:
+            def __init__(self, file, mode):
+                return None
 
-        self._put_setup(fake_destination_file)
-        result = service.put(fake_bytes, "3.bin-e.json")
+            def __enter__(self):
+                return fake_destination_file
 
-        assert result is None
-        assert local.os.path.join.calls == [
-            pretend.call(service._path, "3.bin-e.json"),
-        ]
-        expected_file_path = os.path.join(service._path, "3.bin-e.json")
-        assert local.os.open.calls == [pretend.call(expected_file_path, 0, 0)]
-        assert local.os.fdopen.calls == [pretend.call(0, "wb")]
-        assert fake_destination_file.write.calls == [pretend.call(fake_bytes)]
-        assert fake_destination_file.flush.calls == [pretend.call()]
-        assert fake_destination_file.fileno.calls == [pretend.call()]
-        assert local.os.fsync.calls == [pretend.call("fileno")]
+            def __exit__(self, type, value, traceback):
+                pass
 
-    def test_put_without_restrict(self):
-        service = local.LocalStorage("/path")
+        monkeypatch.setitem(local.__builtins__, "open", FakeDestinationFile)
 
-        fake_destination_file = pretend.stub(
-            write=pretend.call_recorder(lambda *a: None),
-            flush=pretend.call_recorder(lambda: None),
-            fileno=pretend.call_recorder(lambda: "fileno"),
-        )
-
-        fake_bytes = b"data"
-
-        self._put_setup(fake_destination_file, False)
-        result = service.put(fake_bytes, "3.bin-e.json", False)
+        result = service.put(fake_file_data, b"3.bin-e.json")
 
         assert result is None
         assert local.os.path.join.calls == [
-            pretend.call(service._path, "3.bin-e.json"),
+            pretend.call(service._path, b"3.bin-e.json"),
         ]
-        expected_file_path = os.path.join(service._path, "3.bin-e.json")
-        assert local.os.open.calls == [pretend.call(expected_file_path, 0)]
-        assert local.os.fdopen.calls == [pretend.call(0, "wb")]
-        assert fake_destination_file.write.calls == [pretend.call(fake_bytes)]
+        assert fake_destination_file.write.calls == [
+            pretend.call(fake_file_data)
+        ]
         assert fake_destination_file.flush.calls == [pretend.call()]
-        assert fake_destination_file.fileno.calls == [pretend.call()]
+        assert fake_destination_file.flush.calls == [pretend.call()]
         assert local.os.fsync.calls == [pretend.call("fileno")]
 
     def test_put_OSError(self):
@@ -263,17 +225,14 @@ class TestLocalStorageService:
                 )
             ),
             open=pretend.raiser(PermissionError("don't want this message")),
-            O_WRONLY=0,
-            O_CREAT=0,
         )
 
-        local.stat = pretend.stub(S_IRUSR=0, S_IWUSR=0)
         fake_bytes = b"data"
 
-        with pytest.raises(OSError) as err:
+        with pytest.raises(local.StorageError) as err:
             service.put(fake_bytes, "3.bin-e.json")
 
-        assert "don't want this message" in str(err)
+        assert "Can't write role file '/path/3.bin-e.json'" in str(err)
         assert local.os.path.join.calls == [
             pretend.call(service._path, "3.bin-e.json"),
         ]

--- a/tests/unit/tuf_repository_service_worker/services/storage/test_local.py
+++ b/tests/unit/tuf_repository_service_worker/services/storage/test_local.py
@@ -212,7 +212,7 @@ class TestLocalStorageService:
             pretend.call(fake_file_data)
         ]
         assert fake_destination_file.flush.calls == [pretend.call()]
-        assert fake_destination_file.flush.calls == [pretend.call()]
+        assert fake_destination_file.fileno.calls == [pretend.call()]
         assert local.os.fsync.calls == [pretend.call("fileno")]
 
     def test_put_OSError(self):

